posts/web-animation-tips.md

```diff
@@ -118,23 +150,86 @@ Test your animations on a popular low-end device.
+To see this in action, visit my pen [Hiding: visually hidden until focused](https://codepen.io/fuzzylogicx/debug/bGRjvVj) and press the <kbd>tab</kbd> key.
 
 ### Animated disclosure
 
-Like [BBC Sounds’ playlists](https://www.bbc.co.uk/sounds/play/m0012d93) on narrow viewports.
+Here’s an interesting conundrum.
 
-To be written.
+For disclosure (i.e. show and hide) widgets, I tend to either use the native HTML `<details>` element if possible or else a [simple, accessible DIY disclosure](https://codepen.io/fuzzylogicx/pen/YzQjyoj?editors=1010) in which executing a `trigger` toggles a content element’s `hidden` attribute. In both cases, there’s no animation; the change from hidden to revealed and back again is immediate.
+
+To my mind it’s generally preferable to keep it simple and avoid animating a disclosure widget. For a start, it’s tricky! The `<details>` element can’t be (easily) animated. And if using a DIY widget it’ll likely involve animating one of the [expensive properties](#expensive-properties). Animating `height` or `max-height` is also gnarly when working with variable (auto) length content and often requires developers to go beyond CSS and reach for JavaScript to calculate computed element heights. Lastly, forgetting the technical challenges, there’s often _no real need_ to animate disclosure; it might only hinder rather than help the user experience.
+
+But let’s just say you _have to do it_, perhaps because the design spec requires it (like in [BBC Sounds’ expanding and collapsing tracklists](https://www.bbc.co.uk/sounds/play/m0012d93) when viewed on narrow screens).
 
-<!-- this might be relevant https://css-tricks.com/newsletter/277-animating-height/ -->
+Options:
+- [Animate the `<details>` element](https://css-tricks.com/how-to-animate-the-details-element-using-waapi/). This is a nice, standards-oriented approach. But it might only be viable for when you don’t need to mess with `<details>` appearance too much. We’d struggle to apply very custom styles, or to handle a “show the first few list items but not all” requirement like in the BBC Sounds example;
+- [Animate CSS Grid](https://nemzes.net/posts/animating-height-auto/#the-solutions). This is a nice idea but for now the animation only works in Firefox. It’d be great to just consider it a progressive enhancement so it just depends on whether the animation is deemed core to the experience;
+- [Animate from a max-height of 0 to “something sufficient”](https://codepen.io/fuzzylogicx/pen/wvPMxvm) (my pen is inspired by [Scott O’Hara’s disclosure example](https://codepen.io/scottohara/pen/ybLMOm)). This is workable but not ideal; you kinda need to set a max-height sweetspot otherwise your animation will be delayed and too long . You could of course add some JavaScript to get the exact necessary height then set it. BBC use `max-height` for their tracklist animation and those tracklists likely vary in length so I expect they use some JavaScript for height calculation.
```

Comment:
```suggestion
- [Animate from a max-height of 0 to “something sufficient”](https://codepen.io/fuzzylogicx/pen/wvPMxvm) (my pen is inspired by [Scott O’Hara’s disclosure example](https://codepen.io/scottohara/pen/ybLMOm)). This is workable but not ideal; you kinda need to set a max-height sweetspot otherwise your animation will be delayed and too long. You could of course add some JavaScript to get the exact necessary height then set it. BBC use `max-height` for their tracklist animation and those tracklists likely vary in length so I expect they use some JavaScript for height calculation.
```